application/controllers/admin/dumpdb.php

```diff
@@ -67,6 +72,48 @@ private function outputHeaders(string $sFileName)
     {
         header('Content-type: application/octet-stream');
         header('Content-Disposition: attachment; filename=' . $sFileName);
+        
         header("Cache-Control: no-store, no-cache, must-revalidate");  // Don't store in cache because it is sensitive data
     }
+
+    private function getData()
+    {
+        if ($this->data === []) {
+            Yii::app()->loadHelper("admin/backupdb");
+            $dbSize = getDatabaseSize();
+            $downloadable = true;
+            if ($dbSize > Yii::app()->getConfig('maxDatabaseSizeForDump')) {
+                $downloadable = false;
+            }
+            return [
+                'downloadable' => $downloadable,
+                'dbSize' => $dbSize,
+            ];
+        }
+        return $this->data;
+    }
+
+    public function outPutDatabase()
+    {
+        // Check if it's a POST request
+        if (!Yii::app()->request->isPostRequest) {
+            throw new CHttpException(405, gT("Invalid action"));
+        }
+
+        // Check if user has necessary permissions
+        if (!Permission::model()->hasGlobalPermission('superadmin', 'read')) {
+            throw new CHttpException(403, gT("You do not have permission to access this page."));
+        }
+
+        if ($this->data['downloadable'] === false) {
+            throw new CHttpException(403, gT("The database is too large to be downloaded. Please consider exporting it manually using your database client."));
```

Comment:
If not downloadable : why not save it locally (mybb backup for example save it locally and allow dowload via GUI after)